libs/@local/hash-subgraph/src/types/element.ts

```diff
@@ -5,41 +5,48 @@ import {
   PropertyType,
   VersionedUri,
 } from "@blockprotocol/type-system";
-import {
-  DataTypeWithMetadata as DataTypeWithMetadataGraphApi,
-  EntityTypeWithMetadata as EntityTypeWithMetadataGraphApi,
-  ExternalOntologyElementMetadata,
-  OntologyElementMetadata,
-  OwnedOntologyElementMetadata,
-  PropertyTypeWithMetadata as PropertyTypeWithMetadataGraphApi,
-  ProvenanceMetadata as ProvenanceMetadataGraphApi,
-} from "@local/hash-graph-client";
+import { ProvenanceMetadata as ProvenanceMetadataGraphApi } from "@local/hash-graph-client";
 
-import { EntityEditionId, EntityId, EntityVersion } from "./identifier";
+import {
+  EntityId,
+  EntityRecordId,
+  EntityVersion,
+  OntologyTypeRecordId,
+  Timestamp,
+} from "./identifier";
 
 // Due to restrictions with how much OpenAPI can express, we patch the schemas with the better-typed ones from the
 // type-system package.
 
-export type DataTypeWithMetadata = Omit<
-  DataTypeWithMetadataGraphApi,
-  "schema"
-> & { schema: DataType };
+export type OwnedOntologyElementMetadata = {
+  recordId: OntologyTypeRecordId;
+  ownedById: string;
+  provenance: ProvenanceMetadataGraphApi;
+};
+
+export type ExternalOntologyElementMetadata = {
```

Comment:
We also have `provenance` here to track who triggered that record creation in the database

Reply:
Have added, thank you